.argo-ci/ci.yaml

```diff
@@ -0,0 +1,158 @@
+    container:
+      image: argoproj/argocd-cli:v0.7.1
+      command: [sh, -c]
+      args: ["{{inputs.parameters.cmd}}"]
+      env:
+      - name: ENV
+        value: "{{inputs.parameters.env}}"
+      - name: ARGO_CD_APP_PREFIX
+        value: "{{workflow.parameters.argo-cd-app-prefix}}"
+      - name: ARGO_CD_SERVER
+        value: "{{workflow.parameters.argo-cd-server}}"
+      - name: ARGO_CD_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: argo-cd-token
+            key: auth-token
+      - name: ARGO_CD_SYNC_TIMEOUT
+        value: "{{workflow.parameters.argo-cd-sync-timeout}}"
+  - name: ksonnet
+    inputs:
+      parameters:
+      - name: env
+      - name: cmd
+      artifacts:
+      - name: git-sha
+        path: "/tmp/git_sha.txt"
+      - name: code
+        path: "/go/src/github.com/{{workflow.parameters.config-org}}/{{workflow.parameters.config-repo}}"
+        git:
+          repo: "https://github.com/{{workflow.parameters.config-org}}/{{workflow.parameters.config-repo}}.git"
+          revision: "{{workflow.parameters.config-revision}}"
+    container:
+      image: argoproj/argo-cd-tools:latest
+      command: [sh, -c]
+      args: ["{{inputs.parameters.cmd}}"]
+      workingDir: "/go/src/github.com/{{workflow.parameters.config-org}}/{{workflow.parameters.config-repo}}"
+      volumeMounts:
+      - name: git-ssh-key
+        mountPath: "/root/.ssh"
+      env:
+      - name: ENV
+        value: "{{inputs.parameters.env}}"
+      - name: ARGO_CD_APP_PREFIX
+        value: "{{workflow.parameters.argo-cd-app-prefix}}"
+      - name: CONFIG_ORG
+        value: "{{workflow.parameters.config-org}}"
+      - name: CONFIG_REPO
+        value: "{{workflow.parameters.config-repo}}"
+      - name: IMAGE_NAME
+        value: "{{workflow.parameters.image-name}}"
+  - name: ci-dind
+    inputs:
+      parameters:
+      - name: cmd
+      - name: download-docker
+        default: curl -O https://get.docker.com/builds/Linux/x86_64/docker-1.13.1.tgz && tar -xzf docker-1.13.1.tgz && mv docker/docker /usr/local/bin/docker && chmod 755 /usr/local/bin/docker
+      artifacts:
+      - name: code
+        path: "/go/src/github.com/{{workflow.parameters.app-org}}/{{workflow.parameters.app-repo}}"
+        git:
+          repo: "https://github.com/{{workflow.parameters.app-org}}/{{workflow.parameters.app-repo}}.git"
+          revision: "{{workflow.parameters.app-revision}}"
+    container:
+      image: golang:1.10.3
+      command: [sh, -c]
+      args: ["{{inputs.parameters.download-docker}}; until docker ps; do sleep 3; done && {{inputs.parameters.cmd}}"]
+      workingDir: "/go/src/github.com/{{workflow.parameters.app-org}}/{{workflow.parameters.app-repo}}"
+      volumeMounts:
+      - name: docker-cred
+        mountPath: "/root/.docker"
+        readOnly: true
+      env:
+      - name: DOCKER_HOST
+        value: 127.0.0.1
+    outputs:
+      artifacts:
+      - name: git-sha
+        path: /tmp/git_sha.txt
```

Comment:
Instead of an output artifact, lets convert this to an output parameter, who's value is obtained from a file. This way we wont depend on artifact feature at all.

Reply:
fixed